gnd/src/main/java/com/google/android/gnd/persistence/geojson/GeoJsonParser.java

```diff
@@ -48,6 +48,32 @@
     this.uuidGenerator = uuidGenerator;
   }
 
+  /**
+   * Returns the immutable list of tiles specified in {@param geojson} that intersect {@param
+   * bounds}.
+   */
+  public ImmutableList<TileSource> intersectingTiles(LatLngBounds bounds, File file) {
+    try {
+      String fileContents = FileUtils.readFileToString(file, Charset.forName(JSON_SOURCE_CHARSET));
+      // TODO: Separate parsing and intersection checks, make asyc (single, completable).
+      JSONObject geoJson = new JSONObject(fileContents);
+      // TODO: Make features constant.
+      JSONArray features = geoJson.getJSONArray(FEATURES_KEY);
+
+      return stream(toArrayList(features))
+          .map(GeoJsonTile::new)
+          .filter(tile -> tile.boundsIntersect(bounds))
+          .map(this::jsonToTileSource)
+          .map(tileSource -> tileSource.toBuilder().setAreaCount(1).build())
```

Comment:
then here you could just call `TileSource::incrementAreaCount`

Reply:
done, thanks!